File: P2/Ejercicio1_MantenimientoFlutter/lib/filters/EmailDomainFilter.dart
import '../interfaces/Filter.dart';

// filtro que valida el dominio del email
class EmailDomainFilter implements Filter {
  static const allowedDomains = ["gmail.com", "hotmail.com"];

  @override
  String? execute(Map<String, String> request) {
    final email = request['email'] ?? '';
    final parts = email.split('@');
    
    if (parts.length < 2 || !allowedDomains.contains(parts[1])) {
      return "Dominio de correo no permitido";
    }
    return null;
  }
}

File: P1/Ejercicio4/filters/email_domain_filter.rb
# filtro que valida que el dominio del correo sea permitido
# implementa la interfaz Filter
class EmailDomainFilter
  include Filter

  # lista de dominios permitidos
  ALLOWED_DOMAINS = ["gmail.com", "hotmail.com"]

  def execute(request)
    email = request[:email]

    # separa la parte del dominio (después del '@')
    _, domain = email.split("@")

    # comprueba si el dominio está dentro de los permitidos
    unless ALLOWED_DOMAINS.include?(domain)
      raise ValidationError, "Dominio no permitido"
    end
  end
end